java/java.source/src/org/netbeans/modules/java/JavaNode.java

```diff
@@ -213,6 +216,48 @@ protected final Sheet createSheet () {
         });
         sheet.put(ps);
         
+        Project parentProject = FileOwnerQuery.getOwner(super.getDataObject().getPrimaryFile());
+        String fileName = super.getDataObject().getPrimaryFile().getName();
+        // If any of the parent folders is a project, this menu won't appear.
+        if (parentProject == null) {
+            Node.Property arguments = new org.openide.nodes.PropertySupport.ReadWrite<String> (
+                    "runFileArguments", // NOI18N
+                    String.class,
+                    "Arguments",
+                    "Arguments passed to the main method while running the file."
+                ) {
+                    public String getValue () {
+                        return NbPreferences.forModule(JavaNode.class).get(fileName + "_SINGLE_FILE_RUN_ARGUMENTS", "");
```

Comment:
`dataObject.getPrimaryFile().setAttribute(...)` or `getAttribute()` would be my preferred location to store these metadata. These two methods were invented exactly for the purpose of associating additional metadata with files.

Reply:
Great tip! I wasn't aware of this.